btclightclient/spv.go

```diff
@@ -33,38 +33,38 @@ func SPVProofFromHex(txoutProof string, txID string) (*SPVProof, error) {
 
 	pmt, err := PartialMerkleTreeFromHex(txoutProof[160:])
 	if err != nil {
-		return nil, err; 
+		return nil, err
 	}
-	merkleProof, err := pmt.GetProof(txID)	
+	merkleProof, err := pmt.GetProof(txID)
 	if err != nil {
 		return nil, err
 	}
 
 	return &SPVProof{
-		blockHash:  blockheader.BlockHash(),
-		txId:       txID,
-		txIndex:    merkleProof.transactionIndex,
-		merklePath: merkleProof.merklePath,
+		BlockHash:  blockheader.BlockHash(),
+		TxId:       txID,
+		TxIndex:    merkleProof.transactionIndex,
+		MerklePath: merkleProof.merklePath,
 	}, nil
 }
 
 func (spvProof SPVProof) MerkleRoot() chainhash.Hash {
-	hashValue := &spvProof.merklePath[0]
-	numberSteps := len(spvProof.merklePath)
-	transactionIndex := spvProof.txIndex
+	hashValue := &spvProof.MerklePath[0]
+	numberSteps := len(spvProof.MerklePath)
+	transactionIndex := spvProof.TxIndex
 	for i := 1; i < numberSteps; i++ {
-		if transactionIndex % 2 == 0 {
-			hashValue = HashNodes(hashValue, &spvProof.merklePath[i])
+		if transactionIndex%2 == 0 {
+			hashValue = HashNodes(hashValue, &spvProof.MerklePath[i])
 		} else {
-			hashValue = HashNodes(&spvProof.merklePath[i], hashValue);
+			hashValue = HashNodes(&spvProof.MerklePath[i], hashValue)
 		}
 		transactionIndex /= 2
 	}
 	return *hashValue
 }
 
 func (lc *BTCLightClient) VerifySPV(spvProof SPVProof) SPVStatus {
-	lightBlock := lc.btcStore.LightBlockByHash(spvProof.blockHash)
+	lightBlock := lc.btcStore.LightBlockByHash(spvProof.BlockHash)
```

Comment:
Let's fix the name after review.

Reply:
RPC call won't work if members of `SPVProof` are in small case (not exported) cause `SPVProof` object won't be serializable, hence did the name change.